lib/ingress/ingress-props.ts

```diff
@@ -0,0 +1,44 @@
+export interface ServiceProps {
+  readonly name: string;
+  readonly port: number;
+  readonly namespace: string;
+  readonly weight: number;
+}
+
+export interface IngressProps {
+  /**
+   * Whether this is an internal service (behind VPN).
+   * @default false
+   */
+  readonly internal?: boolean;
+
+  /**
+   * Custom selector labels, they will be merged with the default app, role, and instance.
+   * They will be applied to the workload, the pod and the service.
+   * @default { app: "<app label from chart>", role: "server", instance: "<construct id>" }
+   */
+  readonly selectorLabels?: { [key: string]: string };
```

Comment:
In other constructs, we call this `selectorLabels` because we use these labels for `spec.selector.matchLabels`. In case of Ingress, there is no such concept. But as we may still want to be able to add/override labels, I'd just call this - `labels`
```suggestion
  /**
   * Custom labels, they will be merged with the default app, role, and instance.
   * @default { app: "<app label from chart>", role: "server", instance: "<construct id>" }
   */
  readonly labels?: { [key: string]: string };
```

Also, I think we'll need a separate props for `app`, and `instance` labels. Currently, the Ingress objects are created as part of the app's chart, but this construct will live in a separate chart that will be deployed into another namespace.

What we want to do here, is mirror the labels of the application's Pods, and the Service that exposes those Pods, labels.

For example, we have the following Service deployed to Persona's namespace:

```yml
apiVersion: v1
kind: Service
metadata:
  name: api-service
  namespace: persona-220728
  labels:
    app: persona
    canary: "false"
    environment: production
    instance: api
    managed-by: cdk8s
    region: eu
    role: server
    service: persona-eu
```

So in our Ingress chart, we want to be able to define an object that clearly points to the same thing:

```ts
new Ingress(this, `persona-api`, {
  app: "persona",     // <- we could create Ingresses for multiple apps in the same chart
  instance: "api",    // <- different from construct id above
  labels: {
    canary: "false",  // <- additional label for this case
  },
  hostnames: ["https://users.talis.com"],
  serviceRouting: [
    {
      namespace: "persona-220728",
      name: "api-service",
      port: 80,
      weight: 100,
    }
  ],
});
```

The `service` label can be created from `app`, `environment`, and `region` (we can skip watermark here):
https://github.com/talis/talis-cdk8s-constructs/blob/v4.8.0/lib/talis-chart/talis-chart.ts#L60

The end result should be an Ingress + external Service, created in the ingress namespace, with all the labels matching the Service shown above.